src/x11_utils.rs

```diff
@@ -292,6 +292,15 @@ pub fn parse_u8_list(data: &[u8], list_length: usize) -> Result<(&[u8], &[u8]),
     }
 }
 
+/// Returns a ParseError if the input slice is not finished.
+pub fn check_exhausted(data: &[u8]) -> Result<(), ParseError> {
+    if data.is_empty() {
+        return Ok(());
+    }
+
+    Err(ParseError::ParseError)
+}
```

Comment:
Nope. This should never be needed. The X11 server seems to be fine with "trailing garbage" after a request. We also do not check in reply parsing that we reached the end of the reply.